### PR TITLE
EVG-21011: Remove BugSnag

### PR DESCRIPTION
**Pull request by sophstad**
EVG-21011

### Description

- Remove BugSnag initialization
- Update `leaveBreadcrumb` to use more Sentry-native types
- Will remove BugSnag key from `.env-cmdrc.json` and Evergreen variables once changes are deployed

### Testing

- Remove BugSnag tests

### CODE REVIEW

File: src/utils/errorReporting.ts [hunk 1]
```diff
@@ -90,68 +47,50 @@ enum SentryBreadcrumb {
...
-    case "manual":
-    case "state":
-    default:
-      return SentryBreadcrumb.Default;
+const leaveBreadcrumb = (
+  message: string,
+  metadata: Breadcrumb["data"],
+  type: SentryBreadcrumb
+) => {
+  if (!isProductionBuild()) {
```
[the remark this answers is not in the document]
Comment (khelif96) [reply src/utils/errorReporting.ts:55]: I assume that this is so we get sentry logging for all environments including the staging and beta environments. `isProductionBuild` will be true as long as it is a production build. 

We are limited on the amount of sentry replay playback so we probably wouldn't want it enabled in every environment, hence why we sample on prod.

File: scripts/deploy/deploy.sh [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
 echo "Deploying to S3"
 aws s3 sync build/ s3://"${BUCKET}"/ --acl public-read --follow-symlinks --delete --exclude .env-cmdrc.json 
 echo "Deployed to S3"
 
 # If the above step succeeds, run this step
-echo "Uploading source maps to Bugsnag"
-./scripts/deploy/app-version.sh && node ./scripts/deploy/upload-bugsnag-sourcemaps.js
-echo "Source maps uploaded to Bugsnag"
+echo "Exporting REACT_APP_VERSION variable"
+./scripts/deploy/app-version.sh
```
[the remark this answers is not in the document]
Comment (khelif96) [reply scripts/deploy/deploy.sh:12]: I feel like at some point this was also baked into the index.html but I no longer see the code there. 
We should probably add it back.

[the remark this answers is not in the document]
Comment (sophstad) [reply scripts/deploy/deploy.sh]: I removed the script and added the value to `index.html`!

[the remark this answers is not in the document]
Comment (sophstad) [reply scripts/deploy/deploy.sh]: It's now passed in directly as an environment variable via the `build` script in `package.json`. You can verify that it's present in the `<head>` right now on https://spruce-staging.corp.mongodb.com!

**Pull request merged**